coderd/httpapi/httpapi.go

```diff
@@ -52,8 +52,19 @@ func init() {
 
 // Response represents a generic HTTP response.
 type Response struct {
-	Message string  `json:"message" validate:"required"`
-	Errors  []Error `json:"errors,omitempty" validate:"required"`
+	// Message is for general user-friendly error messages. This message will
+	// be shown at the top/bottom of a form, or in a toast on the UI.
+	Message string `json:"message"`
+	// Internal has the technical error information (err.Error()). These details
+	// might come from external packages and might not be user friendly.
+	// Do not populate this error field with any sensitive information or
+	// any errors that may be a security implication. These details are still
+	// available to more technical users.
+	Internal string `json:"internal"`
+	// Errors are form field-specific friendly error messages. They will be
+	// shown on a form field in the UI. These can also be used to add additional
+	// context if there is a set of errors in the primary 'Message'.
+	Errors []Error `json:"errors,omitempty"`
```

Comment:
I think this will be difficult to understand at a glance how I'd display an error to the user.

Here's an alternative structure that just tweaks the names a bit and adds examples, but could help:

```go
type Response struct {
  // An actionable message that depicts actions the request took. Examples:
  // - "A user has been created."
  // - "Failed to create a user."
  Message string `json:"message"`
  // A debug message that provides further insight into why the action failed.
  // - "database: too many open connections"
  // - "stat: too many open files"
  Detail string `json:"detail"`
}
```

I don't know what to name `Errors`. since it's form-specific I'd lean towards `Validations` maybe?

I'm curious for your take @presleyp, you're a magician with words! 🪄

Reply:
@Kira-Pilot suggested `Verbose` instead of `Internal` and I like that; `Detail` works too.

I like `Validations` better than `Errors` - more specific! Right now I feel like it requires an explanatory comment on the frontend.

But I know @Emyrk doesn't want this PR to be open for long, maybe these tweaks can be a followup.

Reply:
I can change that now, that's easy with IDE tools